src/System.Text.Json/src/System/Text/Json/ConsumeTokenResult.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// This enum captures the tri-state return value when trying to read the
+    /// next JSON token.
+    /// </summary>
+    internal enum ConsumeTokenResult : byte
+    {
+        /// <summary>
+        /// Reached a valid end of token and hence no action is required.
+        /// </summary>
+        Success,
+        /// <summary>
+        /// Observed incomplete data but progressed state partially in looking ahead.
+        /// Return false and roll-back to a previously saved state.
+        /// </summary>
+        IncompleteRollback,
```

Comment:
When I encountered this in code I read it as "the result was that I tried to roll back, but that was incomplete".

Perhaps the triplet here should be { Success, Rollback, Incomplete }.  Or { Success, InitiateRollback, Inconclusive }.

It's not major, but it definitely made me stop and think something like "wait, what?"